atc/exec/check_step_test.go

```diff
@@ -3,260 +3,176 @@ package exec_test
 		})
+	})
 
-		It("creates a container with the correct type and owner", func() {
-			_, _, delegate, actualOwner, actualContainerMetadata, actualContainerSpec, actualResourceTypes := fakeWorker.FindOrCreateContainerArgsForCall(0)
-
-			Expect(actualOwner).To(Equal(owner))
-
-			Expect(actualContainerSpec.ImageSpec).To(Equal(worker.ImageSpec{
-				ResourceType: "some-resource-type",
-			}))
-			Expect(actualContainerMetadata).To(Equal(containerMetadata))
-			Expect(actualContainerSpec.Tags).To(Equal([]string{"some", "tags"}))
-			Expect(actualContainerSpec.TeamID).To(Equal(123))
-			Expect(actualContainerSpec.Env).To(Equal(stepMetadata.Env()))
+	Context("having credentials in a resource type", func() {
+		BeforeEach(func() {
+			resTypes := atc.VersionedResourceTypes{
+				{
+					ResourceType: atc.ResourceType{
+						Source: atc.Source{
+							"some-custom": "((super-secret-source))",
+						},
+					},
+				},
+			}
 
-			Expect(actualResourceTypes).To(Equal(interpolatedResourceTypes))
-			Expect(delegate).To(Equal(fakeDelegate))
+			checkPlan = atc.CheckPlan{
+				Source:                 atc.Source{"some": "super-secret-source"},
+				VersionedResourceTypes: resTypes,
+			}
 		})
 
-		Context("when the timeout cannot be parsed", func() {
+		Context("having cred evaluation failing", func() {
+			var expectedErr error
+
 			BeforeEach(func() {
-				checkPlan.Timeout = "bad-value"
+				expectedErr = errors.New("creds-err")
+
+				fakeCredVarsTracker := new(varsfakes.FakeCredVarsTracker)
+				fakeCredVarsTracker.GetReturns(nil, false, expectedErr)
+
+				fakeDelegate.VariablesReturns(fakeCredVarsTracker)
 			})
 
-			It("fails to parse the timeout and returns the error", func() {
-				Expect(stepErr).To(HaveOccurred())
-				Expect(stepErr).To(MatchError("time: invalid duration bad-value"))
+			It("errors", func() {
+				Expect(err).To(HaveOccurred())
+				Expect(errors.Is(err, expectedErr)).To(BeTrue())
 			})
 		})
+	})
 
-		It("times out after the specified timeout", func() {
-			now := time.Now()
-			ctx, _, _ := fakeResource.CheckArgsForCall(0)
-			deadline, _ := ctx.Deadline()
-			Expect(deadline).Should(BeTemporally("~", now.Add(10*time.Second), time.Second))
+	Context("having a timeout that fails parsing", func() {
+		BeforeEach(func() {
+			checkPlan = atc.CheckPlan{
+				Timeout: "th1s_15_n07_r1gh7",
+			}
 		})
 
-		It("invokes resourceFactory -> NewResource with the correct arguments", func() {
-			actualSource, actualParams, actualFromVersion := fakeResourceFactory.NewResourceArgsForCall(0)
-			Expect(actualSource).To(Equal(checkPlan.Source))
-			Expect(actualParams).To(BeNil())
-			Expect(actualFromVersion).To(Equal(checkPlan.FromVersion))
+		It("errors", func() {
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("invalid duration"))
 		})
+	})
 
-		It("runs the check resource action", func() {
-			Expect(fakeResource.CheckCallCount()).To(Equal(1))
+	Context("with a resonable configuration", func() {
```

Comment:
Do you think it would be valuable to have a test case to ensure the worker.Client receives the correct args ?

Reply:
yeahh, sounds like a good thing to test 😁

Reply:
ef039c956e532d58a8ba9ad53a43eba102a3b5ae